kickstart: Fix --noformat and --useexisting

When using --noformat and --useexisting with an anaconda kickstart
file, the internal format flags must be set such that it the volume
exists and is mountable.  Previously the volume would silently not get
mounted and can cause the install to fail entirely.

---
 pyanaconda/kickstart.py |   18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

--- a/pyanaconda/kickstart.py
+++ b/pyanaconda/kickstart.py
@@ -477,6 +477,15 @@ class LogVolData(commands.logvol.F15_Log
 
             dev.format.mountpoint = self.mountpoint
             dev.format.mountopts = self.fsopts
+            if (type):
+                nformat = getFormat(type,
+                                    mountpoint=self.mountpoint,
+                                    label=self.label,
+                                    fsprofile=self.fsprofile,
+                                    mountopts=self.fsopts)
+                if nformat:
+                    dev.format = nformat
+                    dev.format.exists = True
             self.anaconda.dispatch.skip_steps("partition", "parttype")
             return
 
@@ -789,6 +798,15 @@ class PartitionData(commands.partition.F
 
             dev.format.mountpoint = self.mountpoint
             dev.format.mountopts = self.fsopts
+            if (type):
+                nformat = getFormat(type,
+                                    mountpoint=self.mountpoint,
+                                    label=self.label,
+                                    fsprofile=self.fsprofile,
+                                    mountopts=self.fsopts)
+                if nformat:
+                    dev.format = nformat
+                    dev.format.exists = True
             self.anaconda.dispatch.skip_steps("partition", "parttype")
             return
 
